test/integration/main_test.go

```diff
@@ -183,7 +183,16 @@ func deleteAllPods(t *testing.T, namespace string) {
 	t.Log(errPod)
 }
 
-func checkPodsLogs(t *testing.T, kubeClient *kubernetes.Clientset, message string, newLogsOnly ...bool) {
+func logLineTime(t *testing.T, pattern string) time.Time {
+	str := checkPodsLogs(t, clientset, `\d+:\d+:\d+\.\d+.*`+pattern)
```

Comment:
Maybe it would be a bit safer to read from beginning of the line, like:
`^\S\d{2}\d{2}\s\d+:\d+:\d+\.\d+.*`
to parse pattern:
`I0721 13:37:20.340873`
This could help to skip potential lines which have time in the middle of log message

Reply:
nice catch!
unfortunately, there are no newlines in captured logs, which is quite weird.. so "^" does not work. I made it at least more specific with an additional check. Now there would be problem if log line contained whole different log line, which I don't expect to happen much...

Reply:
It should work now